spec/models/searchgov_url_spec.rb

```diff
@@ -34,6 +33,12 @@
         expect(SearchgovUrl.fetch_required.pluck(:url)).
           to eq %w[http://www.agency.gov/new http://www.agency.gov/outdated]
       end
+
+      it 'includes urls that have been enqueued for reindexing' do
```

Comment:
@peggles2 , for consistency with this scope spec, I'd create an "enqueued for reindex" URL in the `before` block:
```ruby
      before do
        SearchgovUrl.create!(url: 'http://www.agency.gov/new')
        SearchgovUrl.create!(
          url: 'http://www.agency.gov/outdated', last_crawled_at: 1.week.ago, lastmod: 1.day.ago
        )
        SearchgovUrl.create!(
          url: 'http://www.agency.gov/current', last_crawled_at: 1.day.ago, lastmod: 1.week.ago
        )
        SearchgovUrl.create!(
          url: 'http://www.agency.gov/enqueued', last_crawled_at: 1.day.ago, enqueued_for_reindex: true
        )
      end
```
And then update the expectation:
```ruby
      it 'includes new & outdated URLs, & URLs that have been enqueued for reindex' do
        expect(SearchgovUrl.fetch_required.pluck(:url)).
          to eq %w[http://www.agency.gov/new
                          http://www.agency.gov/outdated
                          http://www.agency.gov/enqueued]
      end
```

Reply:
I thought about that but thats not behavior we are looking to see. We are not creating a new URL. We are updating an existing url to be re-indexed.  New URLS are going to be indexed anyway.  Which will give you a false positive when you fetch it.  Plus it will be confusing to the developer that reads this test. because your checking links that were created. I like the way its testing this way because its clear its a url that was updated with enqueued for reindexing and now its showing up to be reindexed when the fetch method is called.

Reply:
I see your point. However, I think we should avoid making extra db calls in specs. This spec has to create and then update the record, which is not what the spec needs to test. 

Also, one problem with this spec is that it doesn't necessarily test that the URL is included in the `fetch_required` scope. The `searchgov_url` record in the spec does not have a `last_crawled_at` or `last_crawl_status` value, so it would be included in the scope anyway; this spec would not have failed before the code was written to make it pass. 

Let's chat about this spec today; I have an idea about refactoring.

Reply:
Yeah we can chat. I think update or create your hitting the db.  Tests are suppose to follow the same behavior a user does. The user will never create a url with enqueued for reindex set to true. That will never happen. A user will update an existing url. So the test needs to reproduce that behavior.  The one your suggesting does not do that.

Reply:
Oh and also I am using an existing url that has already been created.  I'm just updating it.

Reply:
@peggles2 , following up on our chat on Friday: 
If using fixtures will be more trouble than it's worth at this point, I would proceed as I suggested originally. I wouldn't worry that a record won't normally be created in that state, as this spec is not testing any user behavior. Throughout the specs, there are records created with various attributes for testing, which is fine for unit tests.